File: ldd-nat-cross-desktop/src/command/setting.ts
import { CommandResult, SettingInfo } from '@/models/types';
import { invoke } from '@tauri-apps/api/core';

export const getSettingInfo = async (): Promise<CommandResult> => {
  return await invoke('get_setting', {});
};

export const updateSettingInfo = async (
  settingInfo: SettingInfo,
): Promise<CommandResult> => {
  return await invoke('update_setting', { settingInfo });
};

export const resetSetting = async (): Promise<CommandResult> => {
  return await invoke('reset_setting', {});
};

File: ldd-nat-cross-desktop/src-tauri/src/command/setting.rs
use crate::{
    model::command::CommandResult,
    store::{model::setting::SettingInfoDO, setting_dao::SettingInfoDAO},
};

#[tauri::command]
pub fn get_setting() -> CommandResult<Option<SettingInfoDO>> {
    let setting_dao = SettingInfoDAO::new();
    match setting_dao.get(1) {
        Ok(r) => CommandResult::ok_with_msg_data("查询成功", r),
        Err(_) => CommandResult::err("查询失败"),
    }
}

#[tauri::command]
pub fn update_setting(setting_info: SettingInfoDO) -> CommandResult<usize> {
    let setting_dao = SettingInfoDAO::new();
    match setting_dao.update(setting_info) {
        Ok(r) => CommandResult::ok_with_msg_data("更新成功", r),
        Err(_) => CommandResult::err("更新失败"),
    }
}

#[tauri::command]
pub fn reset_setting() -> CommandResult<usize> {
    let setting_dao = SettingInfoDAO::new();
    match setting_dao.reset_data() {
        Ok(r) => CommandResult::ok_with_msg_data("初始化成功", r),
        Err(_) => CommandResult::err("初始化失败"),
    }
}
